CodeGeneration/Sources/Utils/CodeGenerationFormat.swift

```diff
@@ -136,11 +136,14 @@ public class CodeGenerationFormat: BasicFormat {
       }
     }
     decreaseIndentationLevel()
-    if !formattedChildren.isEmpty {
-      formattedChildren[formattedChildren.count - 1] = formattedChildren[formattedChildren.count - 1].with(
-        \.trailingTrivia,
-        indentedNewline
-      )
+    if let lastChild = formattedChildren.last {
+      let nextTokenStartsWithNewline = lastChild.nextToken(viewMode: .all)?.leadingTrivia.first?.isNewline ?? false
```

Comment:
Nitpick, I would use `.sourceAccurate` here because we only care about tokens that will actually be printed (not missing tokens). It doesn’t make any difference in practice because we shouldn’t have any missing tokens in the tree but still.

```suggestion
      let nextTokenStartsWithNewline = lastChild.nextToken(viewMode: .sourceAccurage)?.leadingTrivia.first?.isNewline ?? false
```

Reply:
Sure thing, can you fix that typo (g -> t) so I can commit the suggestion from here?

Reply:
Done